_posts/2020-01-08-#14053.md

```diff
@@ -0,0 +1,93 @@
+
+## Notes
+
+- An _address index_ is an index from addresses or scriptPubKeys to the
+  transactions in the block chain that create outputs to that address or spend
+  outputs from that address.
+
+- Adding an address index is a very common feature request for Bitcoin Core.
+  Being able to lookup transactions from an address is useful for block
+  explorers or other services that need to scan the block chain. Example
+  requests:
+  [1](https://github.com/bitcoin/bitcoin/pull/2802#issuecomment-26712372),
+  [2](https://github.com/bitcoin/bitcoin/pull/14053#issuecomment-558344637)
+
+- There have been several patches and additional software to produce
+  transaction indexes over the years:
+  [Bitcore](https://github.com/bitpay/bitcore),
+  [electrs](https://github.com/romanz/electrs) and
+  [ElectrumX](https://github.com/kyuupichan/electrumx/) are three examples.
+
+- There are also centralized block explorer services that provide an
+  address-to-transactions lookup. These offer convenience at the cost
+  of privacy.
+
+- There have been several attempts to add an address index to Bitcoin Core:
+  - [Add an address-based transaction index by sipa](https://github.com/bitcoin/bitcoin/pull/2802)
+  - [Adds transaction index by address by jmcorgan](https://github.com/bitcoin/bitcoin/pull/3652)
+  - [Add address-based index by rnicoll](https://github.com/bitcoin/bitcoin/pull/6835)
+
+- Address indexes are inherently unscalable since they grow linearly with the
+  size of the block chain. Some contributors worry that offering an address
+  index in Bitcoin Core may encourage businesses to build Bitcoin services on
+  infrastructure that can't scale as the size of the block chain increases.
+  Pieter Wuille expresses this concern in the post from his PR: "I hate making it
+  easy to build infrastructure that relies on a fully-indexed blockchain (which
+  shouldn't be necessary), as it may remove the incentive to build more scalable
+  solutions. On the other hand, in cases where the alternative is relying on a
+  trusted third party, this approach is certainly preferable, and would allow an
+  RPC-based blockexplorer, for example."
+
+- For more discussion on the merits of adding an address index to Bitcoin Core,
+  see the discussion in the [13 Dec 2018 Bitcoin Core IRC
+  meeting](http://www.erisian.com.au/meetbot/bitcoin-core-dev/2018/bitcoin-core-dev.2018-12-13-19.00.log.html#l-271).
+
+- This latest attempt to add an address index builds on the new auxiliary
+  indexes base class introduced in [PR
+  13243](https://github.com/bitcoin/bitcoin/pull/13243). The auxiliary index
+  infrastructure runs asynchronously and uses the CValidationInterface to
+  receive new block data. That means adding new indexes should have minimal
+  impact on performance and low risk of introducing a bug that impacts mainline
+  functionality.
+
+- The new address index code and tests (`index/addrindex.cpp`,
```

Comment:
Since "new" in the previous paragraph referred to PR #13243, I ended up git grepping master to find `index/addrindex.*` before realising that "new" here means in PR #14053. Maybe clarify that we're back to the PR under review here for dummies like me :D

Reply:
Good point. Changed the wording from "The new address index code..." to "The address index code ... added in this PR".